avogadro/qtplugins/selectiontool/selectiontool.cpp

```diff
@@ -72,97 +76,92 @@ QWidget* SelectionTool::toolWidget() const
 
 QUndoCommand* SelectionTool::mousePressEvent(QMouseEvent* e)
 {
-  if (e->button() != Qt::LeftButton || !m_renderer)
+  if (e->button() != Qt::LeftButton || !m_renderer) {
+    m_initSelectionBox = false;
     return nullptr;
+  }
 
   m_drawSelectionBox = false;
+  m_initSelectionBox = true;
   m_start = Vector2(e->pos().x(), e->pos().y());
   m_end = m_start;
-  Identifier hit = m_renderer->hit(e->pos().x(), e->pos().y());
-
-  // If an atom is clicked, accept the event, but don't add it to the atom list
-  // until the button is released (this way the user can cancel the click by
-  // moving off the atom, and the click won't get passed to the default tool).
-  if (hit.type == Rendering::AtomType)
-    e->accept();
-
-  // Remove the global accept to prevent the rectangle selection area from being
-  // rendered as the selection code is not yet in place for the scene.
   e->accept();
-
   return nullptr;
 }
 
 QUndoCommand* SelectionTool::mouseReleaseEvent(QMouseEvent* e)
 {
   // If the click is released on an atom, add it to the list
-  if (e->button() != Qt::LeftButton || !m_renderer)
+  if (e->button() != Qt::LeftButton || !m_renderer || m_doubleClick) {
+    m_doubleClick = false;
     return nullptr;
-
+  }
+  shouldClean(e);
```

Comment:
If you click on a single atom and then click again, I think the expectation is that it selects and then deselects.

As far as I can tell, this clears the selection .. so later on, it doesn't know that the atom was previously selected.

See below.